WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/LegacyInPersonPaymentsMenuViewController.swift

```diff
@@ -4,7 +4,7 @@ import Yosemite
 import Experiments
 import Combine
 
-final class InPersonPaymentsMenuViewController: UIViewController {
+final class LegacyInPersonPaymentsMenuViewController: UIViewController {
```

Comment:
Just one consideration here, is this class really legacy until the new code is in place? I mean, if we have to add a new feature fast, we would add it to this class as the new Swift UI Payments components aren't ready; it might be a bit confusing.
What would be the downsides of just leaving this class with its current name, or just adding a comment on top?

Reply:
I see what you're getting at, but given we've started to rewrite it, I think it is legacy now. The alternative would be to call the other one `NewInPersonPaymentsMenuView`. I don't love renaming files really early in their life, because it can make it difficult to view the early development process in the history. Renaming something right before it's removed is less disruptive.

> if we have to add a new feature fast, we would add it to this class as the new Swift UI Payments components aren't ready; it might be a bit confusing.

Well... conversely it's a good prompt to the person adding the new feature quickly to also add it to the new view, or to at least check whether it should be. A comment might get that across, but is more likely not to be noticed.

> What would be the downsides of just leaving this class with its current name, or just adding a comment on top?

Other than the converse scenario above, we could leave the name of this class alone, because the new class will be `InPersonPaymentsMenu`, so there's no clash. However, for the view model, there would be a clash, and to have the names not match up between the existing VC/VM and the new View/VM, seems like it will add confusion.

So, the reasons I went for using the `Legacy` prefix on both are:
- It lets everyone know that the old code is not long for this world.
- It allows us to keep clear pairs of Views/VMs, without needing to do a rename of the new code early in its life.
- It makes it clear which code we delete when we switch to the new one.

I'll add that the timescale of this project is short, ~ 1 week, and there's no work apart from the Deposits Summary, which I'll be doing, which is planned for the payments menu at the moment.

Reply:
Thanks for the explanation @joshheald! The definition of legacy code is blurry 😅 I still have my concerns about calling it Legacy, but your points make sense, your strategy here is the best we can have for this situation 👍